File: app/src/main/java/com/caramelheaven/lennach/data/datasource/database/LennachDatabase.java
package com.caramelheaven.lennach.data.datasource.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.caramelheaven.lennach.data.datasource.database.dao.BoardDao;
import com.caramelheaven.lennach.models.database.BoardFavouriteDb;

/**
 * Created by CaramelHeaven on 20:43, 23/01/2019.
 */
@Database(entities = {BoardFavouriteDb.class}, version = 1, exportSchema = false)
public abstract class LennachDatabase extends RoomDatabase {
    public abstract BoardDao getBoardDao();
}

File: app/src/main/java/com/caramelheaven/lennach/data/datasource/database/LennachDatabase.kt
package com.caramelheaven.lennach.data.datasource.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase

import com.caramelheaven.lennach.data.datasource.database.dao.BoardDao
import com.caramelheaven.lennach.models.database.BoardDb

/**
 * Created by CaramelHeaven on 16:45, 03/02/2019.
 */
@Database(entities = [BoardDb::class], version = 1, exportSchema = false)
abstract class LennachDatabase : RoomDatabase() {
    abstract val boardDao: BoardDao
}